### PR TITLE
starlark: partial

### PR DESCRIPTION
**Pull request by itayd**
not very different (if at all) from https://docs.python.org/3/library/functools.html#functools.partial.

allows:
```
def dump(*args, **kwargs):
    print(args)
    print(sorted(kwargs.items()))
    return len(args) + len(kwargs)

f = partial(dump, 1, 2, a=4, b=5)

print(f())
print(f(3, c=6))
print(f(1, b=9))
```

### CODE REVIEW

File: runtimes/starlarkrt/internal/bootstrap/bootstrap.star [hunk 1]
```diff
@@ -69,3 +69,10 @@ def activity(fn):
     return struct(
         run=lambda *args, **kwargs: run_activity(fn, *args, **kwargs),
     )
+
+# EXPORT: partial
+def partial(f, *args, **kwargs):
+    def wrapper(*args2, **kwargs2):
```
Comment (tebeka) [inline runtimes/starlarkrt/internal/bootstrap/bootstrap.star:75]: Not sure if starlark has it, but maybe do  what the [wraps](https://docs.python.org/3/library/functools.html#functools.wraps) decorator does? (Copy `__doc__` and other meta)

File: runtimes/starlarkrt/internal/bootstrap/bootstrap.star [hunk 2]
```diff
@@ -69,3 +69,10 @@ def activity(fn):
     return struct(
         run=lambda *args, **kwargs: run_activity(fn, *args, **kwargs),
     )
+
+# EXPORT: partial
+def partial(f, *args, **kwargs):
```
Comment (tebeka) [inline runtimes/starlarkrt/internal/bootstrap/bootstrap.star:74]: Why is that required?

File: runtimes/starlarkrt/internal/bootstrap/bootstrap.star [hunk 3]
```diff
@@ -69,3 +69,10 @@ def activity(fn):
...
+    def wrapper(*args2, **kwargs2):
+        kwargs.update(kwargs2)
```
Comment (tebeka) [inline runtimes/starlarkrt/internal/bootstrap/bootstrap.star:76]: In Python this will mutate the original `kwargs` passed in line 74. It's safer to do a copy.

**Review by tebeka — APPROVED** [review]
Can merge. See my comment on copying `kwargs`

**Comment by efiShtain** [comment]
I don't see any value in this,
workflow developers can handle not using advanced programming languages capabilities like partials
especially where they can just create a closure in starlark and do what ever they want in python

File: runtimes/starlarkrt/internal/bootstrap/bootstrap.star (hunk 1 above)
Comment (itayd) [reply runtimes/starlarkrt/internal/bootstrap/bootstrap.star:75]: huh interesting, but i don't think starlark has it.

File: runtimes/starlarkrt/internal/bootstrap/bootstrap.star (hunk 2 above)
Comment (itayd) [reply runtimes/starlarkrt/internal/bootstrap/bootstrap.star:74]: motivation: https://github.com/autokitteh/autokitteh/pull/97#discussion_r1527618164
